salt/client/ssh/__init__.py

```diff
@@ -526,6 +527,11 @@ def run(self):
         outputter = self.opts.get('output', 'nested')
         for ret in self.handle_ssh():
             host = next(six.iterkeys(ret))
+            final_exit = 0
+            host_ret = ret[host].get('retcode')
```

Comment:
This should be `ret[host].get('retcode', 0)`. If retcode doesn't exist, we should assume success. The without the 0, it will return None if it doesn't exist, and will give a non-zero final_exit.